aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -510,4 +534,48 @@ Tensor& capped_random_cuda_(Tensor& self, int64_t to, Generator* gen) {
   return clamped_random_cuda_(self, 0, to, gen);
 }
 
+Tensor& normal_cuda_(Tensor& self, double mean, double std, Generator* gen) {
+  TORCH_CHECK(std > 0.0, "normal_ expects std > 0.0, but found std=", std);
+  auto iter = TensorIterator::nullary_op(self);
+  normal_kernel_cuda(*iter, mean, std, gen);
+  return self;
+}
+
+Tensor& normal_out_cuda(Tensor& output, const Tensor& mean, double std, Generator* gen) {
+  normal_cuda_(output, 0, std, gen);
+  output.add_(mean);
+  return output;
+}
+
+Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std, Generator* gen) {
+  normal_cuda_(output, 0, 1, gen);
+  auto mean_tensor = at::full({1}, mean, output.options());
+  output = mean_tensor.addcmul_(output, std);
```

Comment:
I'm not sure this is how *_out functions are supposed to work, because it reassigns `output`. @ezyang?

Reply:
All inplace functions always return exactly the same reference as they were passed. So what Syed has done here is harmless, albeit also not necessary.

Reply:
This does not fill the memory of `output` with the correct results, `mean_tensor` contains correct results, output is assigned to it. It's the same pattern as below, so if below is wrong, this is wrong too. Am I missing something?